src/pal/src/thread/process.cpp

```diff
@@ -3035,6 +3039,11 @@ Return
 BOOL 
 InitializeFlushProcessWriteBuffers()
 {
+#if HAVE_SYS_MEMBARRIER && defined(_ARM64_)
```

Comment:
As I have mentioned before, we should attempt to call the syscall and if it returns ENOSYS, fall back to the original way. We should set a global flag here so that the actual FlushProcessWriteBuffers knows whether to call the syscall or not. And we would also want to enable it for all architectures, not just ARM64.